### PR TITLE
[docs] Samples for configuring supported disk_formats

### PR DESCRIPTION
**Pull request by konan-abhi**
Added sample documentation to show how to configure supported disk_formats in deployment.

Fixes: [OSPRH-1232](https://issues.redhat.com//browse/OSPRH-1232)

### CODE REVIEW

File: config/samples/disk_formats/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+For example, use the following configuration to enable only RAW and ISO disk formats:
+
+```
+...
+  glance:
+    template:
+      customServiceConfig: |
+        [image_format]
+         disk_formats=raw,iso
+      glanceAPI:
```
[the remark this answers is not in the document]
Comment (konan-abhi) [reply config/samples/disk_formats/README.md:23]: Done

File: config/samples/disk_formats/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+For example, use the following configuration to reject QCOW2 disk images:
+
+```
+...
+  glance:
+    template:
+      customServiceConfig: |
+        [image_format]
+         disk_formats=raw,iso,aki,ari,ami
```
[the remark this answers is not in the document]
Comment (konan-abhi) [reply config/samples/disk_formats/README.md:37]: Done

File: config/samples/disk_formats/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      glanceAPI:
```
[the remark this answers is not in the document]
Comment (konan-abhi) [reply config/samples/disk_formats/README.md:38]: Done

File: config/samples/disk_formats/README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        ...
+...
+...
+```
+
+## Configuring supported disk-formats
+
+Assuming you are using `install_yamls` and you already have `crc` running, you
```
[the remark this answers is not in the document]
Comment (konan-abhi) [reply config/samples/disk_formats/README.md:46]: Done, I don't know how the changes got reverted :(

File: config/samples/disk_formats/README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+$ make crc_storage openstack
+$ oc kustomize ../glance-operator/config/samples/image-cache > ~/openstack-deployment.yaml
+$ export OPENSTACK_CR=`realpath ~/openstack-deployment.yaml`
+$ make openstack_deploy
+```
+
+If we already have a deployment working we can always use `oc kustomize ../image-cache | oc apply -f -`
+from this directory to make the changes.
+
+## Glance Cache cleaner and pruner utilities
```
[the remark this answers is not in the document]
Comment (konan-abhi) [reply config/samples/disk_formats/README.md:61]: Done, I removed it earlier but :(

File: config/samples/disk_formats/disk-formats.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackControlPlane
+metadata:
+  name: openstack
+spec:
+  glance:
+    template:
+      databaseInstance: openstack
+      serviceUser: glance
+      containerImage: quay.io/podified-antelope-centos9/openstack-glance-api:current-podified
```
[the remark this answers is not in the document]
Comment (konan-abhi) [reply config/samples/disk_formats/disk-formats.yaml:13]: Done

File: config/samples/disk_formats/disk-formats.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      customServiceConfig: |
```
[the remark this answers is not in the document]
Comment (konan-abhi) [reply config/samples/disk_formats/disk-formats.yaml:14]: Done

File: config/samples/disk_formats/disk-formats.yaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        default:
+          debug:
+            service: false
+          preserveJobs: false
+          replicas: 1
+      secret: osp-secret
+      storageClass: ""
+      storageRequest: 1G
+      imageCacheSize: 10G
+  extraMounts:
```
[the remark this answers is not in the document]
Comment (konan-abhi) [reply config/samples/disk_formats/disk-formats.yaml:37]: Done

[the remark this answers is not in the document]
Comment (konan-abhi) [reply config/samples/disk_formats/disk-formats.yaml:37]: Done

File: config/samples/disk_formats/README.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+Each image in Glance has an associated disk format property.
+When creating an image the user specifies a disk format. They must
+select a format from the set that the Glance service supports.
+An operator can add or remove disk formats to the supported set.  This is
+done by setting the ``disk_formats`` parameter which is found in the
+``[image_format]`` section of ``glance-api.conf``.
+
+``disk_formats=<Comma separated list of disk formats>``
+ 
+Default supported formats are; ``ami,ari,aki,vhd,vhdx,vmdk,raw,qcow2,vdi,iso,ploop``
```
Comment (fmount) [inline config/samples/disk_formats/README.md:12]: typo: `s/;/:`

**Review by fmount — COMMENTED** [review]
I only see a typo, otherwise looks good and we can merge after fixing the `pre-commit` check on the trailing space [1]!

[1] https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openstack-k8s-operators_glance-operator/425/pull-ci-openstack-k8s-operators-glance-operator-main-precommit-check/1752253059729199104

File: config/samples/disk_formats/README.md (hunk 9 above)
Comment (konan-abhi) [reply config/samples/disk_formats/README.md:12]: Done

**Comment by fmount** [comment]
[pre-commit failure](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openstack-k8s-operators_glance-operator/425/pull-ci-openstack-k8s-operators-glance-operator-main-precommit-check/1752262197725106176/artifacts/test/build-log.txt) 

```
 trim trailing whitespace.................................................Failed
- hook id: trailing-whitespace 
```

File: config/samples/disk_formats/disk-formats.yaml [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+kind: OpenStackControlPlane
+metadata:
+  name: openstack
+spec:
+  glance:
+    template:
+      databaseInstance: openstack
+      serviceUser: glance
+      customServiceConfig: |
+        [DEFAULT]
```
[the remark this answers is not in the document]
Comment (konan-abhi) [reply config/samples/disk_formats/disk-formats.yaml:12]: Done

File: config/samples/disk_formats/kustomization.yaml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+resources:
```
[the remark this answers is not in the document]
Comment (konan-abhi) [reply config/samples/disk_formats/kustomization.yaml:1]: Done

File: config/samples/disk_formats/kustomization.yaml [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+- ../base/openstack
```
Comment (fmount) [inline config/samples/disk_formats/kustomization.yaml:2]: sorry, I was wrong on L2: it should be `../backends/base/openstack`

**Review by fmount — CHANGES_REQUESTED** [review]

**Comment by fmount** [comment]
@konan-abhi apart from my last comment (sorry, I overlooked the path) looks great!

File: config/samples/disk_formats/kustomization.yaml (hunk 12 above)
Comment (konan-abhi) [reply config/samples/disk_formats/kustomization.yaml:2]: NP, done!

**Review by fmount — APPROVED** [review]
/lgtm

**Pull request merged**